mcs_benchmark_data/cli/commands/_command.py

```diff
@@ -0,0 +1,113 @@
+import logging
+import importlib
+from abc import ABC, abstractmethod
+from configargparse import ArgParser
+from types import FunctionType
+
+from typing import Optional
+from pathlib import Path
+from string import Template
+
+from mcs_benchmark_data.dataset_type import DatasetType
+from mcs_benchmark_data.cli.template_type import TemplateType
+
+
+class _Command(ABC):
+    """
+    A command-line (sub-)command.
+    For each _Command, the command-line entrypoint (main):
+    1) Creates an ArgumentParser sub-command parser
+    2) Calls .add_arguments to add sub-command specific arguments to this sub-parser
+    If the sub-command is invoked, then
+    3) Invoke .__call__ with the parsed arguments
+    """
+
+    def __init__(self):
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    def add_arguments(
+        self, arg_parser: ArgParser, add_parent_args: FunctionType
+    ) -> None:
+        """
+        Add sub-command-specific arguments to the argparse (sub-) ArgParser
+        """
+
+    @abstractmethod
+    def __call__(self, args) -> None:
+        """
+        Invoke .__call__ with the parsed arguments
+        """
+
+    def _create_files_from_template(
+        self,
+        *,
+        root_path: Path,
+        benchmark_name: str,
+        data_dir: str,
+        format_args: dict,
+        submission_name: Optional[str] = None,
+        is_first_submission: Optional[bool] = None,
+    ):
+
+        repo_root = Path(__file__).parent.parent.parent.parent
+
+        path_to_templates = repo_root / "mcs_benchmark_data" / "cli" / "templates"
+
+        for template_type in TemplateType:
+
+            if (
+                template_type == TemplateType.METADATA
+                and is_first_submission is not None

```

Comment:
If is_first_submission is None, then wouldn't `not is_first_submission` return true, which I do not want?

Reply:
You're right, I misread the code. Expecting something else.

The usual pattern with an Optional[bool] parameter is to have unspecified (None) be equivalent to False, so you either set parameter=True or you don't specify the parameter at all (rather than explicitly setting it to False).